CLI: Add url flag to download templates in sb repro

yannbf opened the pull request:
Issue: N/A

## What I did

This PR integrates `degit` into the sb cli.

It adds two features:

### Prefer downloading from storybook repro repository

This way, the CLI just downloads a git repo from [repro-templates](https://github.com/storybookjs/repro-templates) instead of having to run a generator (e.g. create-react-app myApp...). This could save time both in CI but also for the `sb repro` experience.

If the selected template does not exist in repro-templates, it will fallback to the original generator.

For the user, there's no change in the command. Optionally, you could use the `--skip-git` flag to use the generators directly instead.

The repro template has both `main` and `next` branches which old latest vs next versions of Storybook. The script will detect whether the user is using `sb repro` or `sb repro@some-alpha-version` and will download from the correct branch

### Set project via --url flag

`sb repro` now has a `--url` flag which allows you to download any git project (github, gitlab, bitbucket..) and bootstrap Storybook on top of it. Could be useful for QAing purposes in case we find big projects with certain settings that could be valuable to test out. 

This uses `degit` under the hood so you can use all kinds of formats for the URL such as `username/project` or `https://github.com/username/project` or other formats that degit supports.

## How to test

First check the branch, bootstrap it or just build the cli:
```
yarn build cli
```

Next, somewhere in an experimentation folder, run `sb repro` using the URL of a project that does not contain Storybook.

```
# This will prompt for location folder, using the repo as default folder name
../storybook/lib/cli/bin/index.js repro --url the-url-here
```

```
# This will download and install things automatically with no prompt
../storybook/lib/cli/bin/index.js repro --url the-url-here outputfoldername
```

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

<!--

Everybody: Please submit all PRs to the `next` branch unless they are specific to the current release. Storybook maintainers cherry-pick bug and documentation fixes into the `master` branch as part of the release process, so you shouldn't need to worry about this. For additional guidance: https://storybook.js.org/docs/react/contribute/how-to-contribute

Maintainers: Please tag your pull request with at least one of the following:
`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

-->

ghengeveld committed:
Show exceptions by non-instrumented code in the Interactions panel

shilman commented on lib/cli/src/repro-generators/configs.ts:
I think this shouldn't be included in the code. It should just try & fail & run the full generator OR the user should pass `--skip-git`.

shilman reviewed (changes requested):
Looks like this breaks CI. Can you please take a look when you get back from holiday?

yannbf commented on lib/cli/src/repro-generators/configs.ts:
That's the case. I just added that name to make it more apparent, but that flag is set based on --skip-git

yannbf commented:
> Looks like this breaks CI. Can you please take a look when you get back from holiday?

If we want to already have CI download repos we will need to fix the repro-generators script first

AriPerkkio committed:
fix: builder stats typings to be optional

yannbf commented:
Also @shilman here's a small summary of timing changes from before/after. This will decrease a ton as we increase the amount of repros in the repro-templates repository

![image](https://user-images.githubusercontent.com/1671563/172209692-f874d578-5b6e-43b5-8949-1caaf242f0ba.png)

yannbf commented:
@shilman in order for this to work with `future` we'll need a `future` branch in the repro templates. Else this could lead to a situation where people do `npx sb@future repro` and it will actually download from `repro-templates` at `next` branch

ghengeveld committed:
Add story for Panel with caught exception

ghengeveld committed:
Fix spying on nested actions

ghengeveld committed:
Only show yellow warning background for caught exceptions, not intercepted exceptions

ghengeveld committed:
Add aria-label for debugger controls

ghengeveld committed:
Test debugger controls

ghengeveld committed:
Deal with exceptions that aren't an Error

ghengeveld committed:
Flatten components list and pull out InteractionsPanel from Panel

ghengeveld committed:
Pull TabStatus out of Panel

ghengeveld committed:
Remove unused import

ghengeveld committed:
Update tests

ghengeveld committed:
Ignore deepscan error

ghengeveld committed:
Fix serialization of object properties

ghengeveld committed:
Chromatic will fail when we deliberately throw an exception

ghengeveld committed:
Rename stories

ghengeveld committed:
Avoid stacked theme because it'll break interactions

tmeasday committed:
Add `docs` option and use to drive the indexer

tmeasday committed:
Add `docs` option and use to drive the indexer

tmeasday committed:
Added test for disabling

ghengeveld committed:
Fix exception colors in dark theme

ghengeveld committed:
Add unit tests for Panel.getInteractions

ghengeveld committed:
Prevent Interactions panel from picking up exceptions that don't originate from the play function

tmeasday committed:
Update StoryIndexGenerator to generate "template" entries

Template (standalone: false) entries are needed for:
  - Docs Page
  - *.stories.mdx files that define a docs page and stories.

A Template entry expects to be rendered from and against a story file. (A standalone entry expects to be rendered as a MDX file)

tmeasday committed:
Some TS fixes

tmeasday committed:
Exclude test files properly in `core-server`

ghengeveld committed:
Wait for idle before starting play function

ghengeveld committed:
Don't introduce an arbitrary await

ghengeveld committed:
Prevent flicker on the interactions counter

ghengeveld committed:
Avoid running play function if story errored

ndelangen committed:
Merge branch 'future/pbm/stage0' into drop-emotion-10-types

# Conflicts:
#	lib/theming/package.json

ndelangen committed:
fix

ndelangen committed:
Merge branch 'future/pbm/stage0' into drop-emotion-10-types

ndelangen committed:
Merge branch 'future/pbm/stage0' into future/type-checking-script

ndelangen committed:
make theming use TSup again (didn't I do this before?!?!?!

ndelangen committed:
Merge branch 'future/pbm/stage0' into drop-emotion-10-types

ndelangen committed:
Merge branch 'future/pbm/stage0' into future/type-checking-script

# Conflicts:
#	lib/theming/package.json

ndelangen committed:
cleanup

ndelangen committed:
Merge branch 'future/base' into future/pbm/stage0

ndelangen committed:
fix

ndelangen committed:
Merge branch 'future/pbm/stage0' into drop-emotion-10-types

ndelangen committed:
Merge branch 'future/pbm/stage0' into future/type-checking-script

ndelangen committed:
add assert polyfill, urgh! yuck!

ndelangen committed:
add assert polyfill, urgh, yuck!

ndelangen committed:
Merge branch 'future/pbm/stage0' into future/type-checking-script

ndelangen committed:
Merge branch 'future/pbm/stage0' into drop-emotion-10-types

ndelangen committed:
fix

ndelangen committed:
Merge branch 'future/pbm/stage0' into future/type-checking-script

ndelangen committed:
Merge branch 'future/pbm/stage0' into drop-emotion-10-types

ndelangen committed:
..

ndelangen committed:
Merge branch 'future/pbm/stage0' into future/type-checking-script

ndelangen committed:
Merge branch 'future/pbm/stage0' into drop-emotion-10-types

ndelangen committed:
cleanup

ndelangen committed:
swap out assert for a hyper-lightweight alternative. It has what we need in doctrine

ndelangen committed:
try this

tmeasday committed:
Render standalone and template stories

tmeasday committed:
Deepscan + snapshot updates

ndelangen committed:
Merge branch 'future/pbm/stage0' into drop-emotion-10-types

ndelangen committed:
Merge branch 'future/pbm/stage0' into future/type-checking-script

ndelangen committed:
use playright in the bench

ndelangen committed:
Merge pull request #18606 from storybookjs/future/playright-bench

use playright in the bench

ndelangen committed:
Merge pull request #18603 from storybookjs/future/type-checking-script

Build: Add a check script to each package

ndelangen committed:
replace the rest

ghengeveld committed:
Monkey-patch callbacks instead of listening on the channel

ndelangen committed:
fix

tmeasday committed:
Fix e2e tests!

tmeasday committed:
It's not so easy to differentiate single story entries

tmeasday committed:
Ok, I think I fixed all e2e tests

tmeasday committed:
Disable test runner docs test

cc @yannbf

ghengeveld committed:
Better error message

ghengeveld committed:
Fix tests

ndelangen committed:
Merge branch 'next' into future/base

# Conflicts:
#	addons/storyshots/storyshots-puppeteer/package.json
#	addons/storysource/package.json
#	examples/official-storybook/stories/addon-a11y/highlight.stories.js
#	lib/api/src/lib/stories.ts
#	lib/api/src/tests/stories.test.ts
#	lib/core-server/src/build-static.ts
#	lib/core-server/src/utils/StoryIndexGenerator.ts
#	lib/manager-webpack4/src/presets/manager-preset.ts
#	lib/telemetry/src/get-monorepo-type.ts
#	yarn.lock

ndelangen committed:
fix

ndelangen committed:
Merge branch 'next' into future/base-with-next

# Conflicts:
#	addons/storyshots/storyshots-puppeteer/package.json
#	addons/storysource/package.json
#	examples/official-storybook/stories/addon-a11y/highlight.stories.js
#	lib/api/src/lib/stories.ts
#	lib/api/src/tests/stories.test.ts
#	lib/blocks/src/controls/Text.tsx
#	lib/core-server/src/build-static.ts
#	lib/core-server/src/utils/StoryIndexGenerator.ts
#	lib/manager-webpack4/src/presets/manager-preset.ts
#	lib/telemetry/src/get-monorepo-type.ts
#	yarn.lock

ndelangen committed:
Merge pull request #18617 from storybookjs/future/base-with-next

Future/base with next

ndelangen committed:
Merge branch 'future/base' into future/prebuilt-manager-esbuild

# Conflicts:
#	lib/manager-webpack5/src/presets/manager-preset.ts

ndelangen committed:
Merge branch 'future/pbm/stage0' of github.com:storybookjs/storybook into future/prebuilt-manager-esbuild

# Conflicts:
#	addons/storyshots/storyshots-puppeteer/package.json
#	yarn.lock

ndelangen committed:
Merge branch 'future/prebuilt-manager-esbuild' into future/drop-emotion-10-types

ghengeveld committed:
Ignore the stacktrace in Chromatic snapshots

ndelangen committed:
Merge branch 'next' into future/base

# Conflicts:
#	examples/vue-kitchen-sink/package.json
#	lib/api/src/index.tsx
#	lib/api/src/modules/stories.ts
#	lib/cli/src/frameworks/html/ts/Button.stories.ts
#	lib/cli/src/frameworks/html/ts/Header.stories.ts
#	lib/cli/src/frameworks/html/ts/Page.stories.ts
#	lib/core-client/src/typings.d.ts
#	lib/core-common/src/types.ts
#	lib/core-server/package.json
#	lib/core-server/src/utils/open-in-browser.ts
#	lib/manager-webpack4/typings.d.ts
#	yarn.lock

shilman committed:
7.0.0-alpha.9 changelog

shilman committed:
Update root, peer deps, version.ts/json to 7.0.0-alpha.9 [ci skip]

shilman committed:
v7.0.0-alpha.9

shilman committed:
Update git head to 7.0.0-alpha.9, update yarn.lock

shilman committed:
7.0.0-alpha.10 changelog

shilman committed:
Update root, peer deps, version.ts/json to 7.0.0-alpha.10 [ci skip]

shilman committed:
v7.0.0-alpha.10

shilman committed:
Update git head to 7.0.0-alpha.10, update yarn.lock

ndelangen committed:
Merge branch 'next' into future/base

ndelangen committed:
Merge branch 'future/base' into future/prebuilt-manager-esbuild

# Conflicts:
#	addons/essentials/package.json
#	frameworks/html-webpack5/package.json
#	frameworks/preact-webpack5/package.json
#	frameworks/react-webpack5/package.json
#	frameworks/server-webpack5/package.json
#	frameworks/svelte-webpack5/package.json
#	frameworks/vue-webpack5/package.json
#	frameworks/vue3-webpack5/package.json
#	frameworks/web-components-webpack5/package.json
#	lib/blocks/package.json
#	lib/cli/src/versions.ts
#	lib/components/package.json
#	lib/core-common/package.json
#	lib/core-server/package.json
#	lib/core-server/src/utils/open-in-browser.ts
#	lib/manager-webpack5/package.json
#	lib/manager-webpack5/typings.d.ts
#	lib/theming/package.json
#	lib/ui/package.json
#	presets/html-webpack/package.json
#	presets/preact-webpack/package.json
#	presets/react-webpack/package.json
#	presets/server-webpack/package.json
#	presets/svelte-webpack/package.json
#	presets/vue-webpack/package.json
#	presets/vue3-webpack/package.json
#	presets/web-components-webpack/package.json
#	renderers/html/package.json
#	renderers/preact/package.json
#	renderers/react/package.json
#	renderers/server/package.json
#	renderers/svelte/package.json
#	renderers/vue/package.json
#	renderers/vue3/package.json
#	renderers/web-components/package.json
#	yarn.lock

Andarist committed:
Make Emotion's `Theme` augmentation work in prod and dev

tmeasday committed:
Refactor DocsRender/Context

tmeasday committed:
Refactor ExternalX to reuse DocsContext

Simplifies things down a lot!

ndelangen committed:
no unsafe eval

ndelangen committed:
add support for preview-head.html to new prebuild manager
add support for setting config after manager has loaded

ndelangen committed:
add refs feature back

ndelangen committed:
fix test

ndelangen committed:
fix deepscan issue

ndelangen committed:
improvements to setup

ndelangen committed:
fix linting

ndelangen committed:
Merge pull request #18633 from storybookjs/future/base-improved

Future - base improved

ndelangen committed:
merge future/base

ndelangen committed:
cleanup

ndelangen committed:
fix removed new package

tmeasday committed:
Don't append title prefix to Meta of= docs entries

tmeasday committed:
Add some template exampes to external docs

And make sure doc blocks use the "primary" story by id when possible.

ndelangen committed:
add comments to explain the situation with globals, and why we do what we do.

tmeasday committed:
Code review

ghengeveld committed:
Merge pull request #16592 from storybookjs/16585-exception-handling

Interactions: Show exceptions by non-instrumented code in panel

ndelangen committed:
cleanup

ndelangen committed:
cleanup and add LOGLEVEL support

ndelangen committed:
fix and add a CI check that generated files are not mutated during CI run

ndelangen committed:
move event name to core-events package

ndelangen committed:
Merge pull request #18550 from storybookjs/future/pbm/stage0

Future / pre-built manager - using esbuild

ndelangen committed:
fix

ndelangen committed:
fix

ndelangen committed:
add a linter for .ejs files (our templates)

ndelangen committed:
deduplicate interopRequireDefault

ndelangen committed:
change interpret for esbuild-register and make loadAllPresets async

ndelangen committed:
Merge branch 'future/prep-for-esbuild-register' into future/async-load-presets-esbuild-register

# Conflicts:
#	lib/core-common/src/presets.ts
#	lib/core-common/src/utils/interpret-require.ts

ndelangen committed:
register just once, do not unregister

ndelangen committed:
Merge pull request #18638 from storybookjs/future/prep-for-esbuild-register

Future - prep for #18619

ndelangen committed:
Merge pull request #18637 from storybookjs/future/linting-ejs

Future - add a linter for .ejs files (our templates)

ndelangen committed:
Merge branch 'future/base' into future/drop-emotion-10-types

# Conflicts:
#	scripts/prepare/bundle.ts

ndelangen committed:
a bunch of fixes

ndelangen committed:
cleanup

ndelangen committed:
cleanup

ndelangen committed:
cleanup

shilman committed:
7.0.0-alpha.11 changelog

shilman committed:
Update root, peer deps, version.ts/json to 7.0.0-alpha.11 [ci skip]

shilman committed:
v7.0.0-alpha.11

shilman committed:
Update git head to 7.0.0-alpha.11, update yarn.lock

ndelangen committed:
Merge branch 'future/base' into luke/next

# Conflicts:
#	examples/web-components-kitchen-sink/yarn.lock
#	lib/api/src/modules/layout.ts
#	lib/blocks/src/blocks/SourceContainer.tsx

ndelangen committed:
Merge branch 'future/base' into fix/builder-stats-typings

ndelangen committed:
fix lint error

ndelangen committed:
only show errors in lint output, scrolling past 10000 warnings isn't helping

ndelangen committed:
Merge pull request #18608 from storybookjs/luke/next

Future - dependencies: use `dequal` for equality checks

ndelangen committed:
Merge pull request #18642 from storybookjs/future/improve-linting-a-bit

Future - improve linting a bit

ghengeveld committed:
Ignore tooltips in Chromatic

ndelangen committed:
Merge branch 'next' into future/base-with-next-merged

# Conflicts:
#	frameworks/ember/src/client/preview/render.ts
#	lib/components/package.json
#	yarn.lock

ndelangen committed:
Merge pull request #18647 from storybookjs/future/base-with-next-merged

Future - merge next into future/base

ndelangen committed:
Merge pull request #18377 from AriPerkkio/fix/builder-stats-typings

Core: Fix builder stats typings to be optional

ghengeveld committed:
Remove chromatic ignore attributes, instead just don't run the play function

tmeasday committed:
Refactor for clarity as per suggestion

tmeasday committed:
Update snapshots

tmeasday committed:
Properly disable `e2e-tests-sb-docs`

tmeasday committed:
Properly disable `e2e-tests-sb-docs`

tmeasday committed:
Deepscan

ndelangen committed:
Merge branch 'next' into future/base

# Conflicts:
#	lib/cli/src/generators/baseGenerator.ts
#	renderers/svelte/src/docs/extractArgTypes.ts
#	renderers/svelte/src/render.ts

shilman committed:
Merge pull request #18573 from storybookjs/tom/sb-450-index-docspage-processing-configuration

Addon-docs: Add docs index configuration via main.js

shilman committed:
Merge pull request #18574 from storybookjs/tom/sb-449-index-update-generation-to-produce

Addon-docs: Produce docs page entries in the index

tmeasday committed:
Rebased content got lost

tmeasday committed:
Get rid of identifiers from `DocsContext`

tmeasday committed:
Cleanup

tmeasday committed:
Deepscan Cleanup

shilman committed:
Merge pull request #18619 from storybookjs/future/async-load-presets-esbuild-register

Core: Async load presets, replace interpret with esbuild-register

shilman committed:
7.0.0-alpha.12 changelog

ghengeveld committed:
Document the instrumenter

ghengeveld committed:
Remove ambiguity

ghengeveld committed:
Add note about forceRemount

ghengeveld committed:
Add Options section

ghengeveld committed:
Move Options up to API

shilman committed:
Merge pull request #18595 from storybookjs/tom/sb-451-preview-no-longer-allow-story-entries-to

Preview: Handle new docs-page index entries

shilman committed:
Update CHANGELOG.md

shilman committed:
Merge branch 'future/base' of github.com:storybookjs/storybook into future/base

shilman committed:
Update CHANGELOG.md

shilman committed:
Update root, peer deps, version.ts/json to 7.0.0-alpha.12 [ci skip]

shilman committed:
v7.0.0-alpha.12

shilman committed:
Update git head to 7.0.0-alpha.12, update yarn.lock

shilman committed:
Merge pull request #18634 from storybookjs/no-title-prefix-for-meta-of

Index: Don't prepend `titlePrefix` to a docs entry that references a CSF file's title

ghengeveld committed:
Lots of details

ghengeveld committed:
Details about emitted events

ghengeveld committed:
Merge pull request #18665 from storybookjs/document-instrumenter

Document `@storybook/instrumenter`

tmeasday committed:
Clean up debugging

ghengeveld committed:
Typing for event payload

tmeasday committed:
Update tests

tmeasday committed:
Fix issue with angular docs

tmeasday committed:
Primary doc block should use `storyById()`

tmeasday committed:
Remove docs tab and concept of `docsOnly` from ui

ghengeveld committed:
Merge pull request #18674 from storybookjs/sync-payload-type

Instrumenter: `SyncPayload` type for `sync` event

tmeasday committed:
Some tweaks

shilman committed:
Addon-docs: Delete register.js

tmeasday committed:
Merge pull request #18635 from storybookjs/tom/sb-452-blocks-update-blocks-to-work-in-docspage

Addon-dcos: Refactor DocsRender/Context

tmeasday committed:
Drop race condition

tmeasday committed:
Entry can be undefined when booting up

shilman committed:
Merge branch 'tom/sb-430-ui-remove-docs-tab' of github.com:storybookjs/storybook into tom/sb-430-ui-remove-docs-tab

tmeasday committed:
Deepscan

tmeasday committed:
Fix cypress

tmeasday committed:
Merge remote-tracking branch 'origin/future/base' into tom/sb-430-ui-remove-docs-tab

tmeasday committed:
Update preview stories

tmeasday committed:
Merge pull request #18677 from storybookjs/tom/sb-430-ui-remove-docs-tab

Remove docs tab

shilman committed:
7.0.0-alpha.13 changelog

shilman committed:
Update root, peer deps, version.ts/json to 7.0.0-alpha.13 [ci skip]

shilman committed:
v7.0.0-alpha.13

shilman committed:
Update git head to 7.0.0-alpha.13, update yarn.lock

tmeasday committed:
Move `DocsRenderer` back to `addon-docs`

tmeasday committed:
Move MDX dependency from blocks and cleanup theme

tmeasday committed:
Update to mdx-csf canaries

tmeasday committed:
Fixes

tmeasday committed:
Update mdx1

tmeasday committed:
No longer reset components in `Story` block

It shouldn't be necessary as a different `React` instance renders the user's component.

tmeasday committed:
Export `DocsRenderer` from addon-docs

tmeasday committed:
Don't clean in watch mode.

cc @ndelangen this fixes annoying errors like:

```
error - ../../lib/blocks/dist/index.js
Module build failed: Error: ENOENT: no such file or directory, open '/Users/tom/GitHub/storybookjs/storybook/lib/blocks/dist/index.js'
```

tmeasday committed:
Ensure each preview has its own channel

tmeasday committed:
Fix unsued imports

tmeasday committed:
Fixes

shilman committed:
Merge branch 'next' into future/base

shilman committed:
Merge pull request #18723 from storybookjs/future/update-from-next-7-15

Update future/base from next

tmeasday committed:
Set other manager-side constants in build

tmeasday committed:
Only load docs entries into the hash in docsMode

tmeasday committed:
No need for `collapse` any more!

tmeasday committed:
Remove (almost all) references to `DOCS_MODE` directly

tmeasday committed:
Remove last use of `DOCS_MODE`

tmeasday committed:
Unused import

tmeasday committed:
Pass docsMode from refs too

tmeasday committed:
Update MDX stories to use `MDXProvider`

tmeasday committed:
Simplify channel use in blocks

tmeasday committed:
Fix TS type error

tmeasday committed:
Fixes

tmeasday committed:
Fix stories that need channel

ndelangen committed:
Merge pull request #18728 from storybookjs/export-other-constants-for-manager

Set other manager-side constants in build

shilman committed:
Merge pull request #18729 from storybookjs/tom/sb-327-ui-sort-out-docs_mode

Preview: Simplify docsMode

shilman committed:
Merge pull request #18709 from storybookjs/tom/sb-465-remove-call-to-addtemplate-from-mdx

Remove `AddContext` from mdx packages

tmeasday committed:
Rename `parameters` to `docsParameter` as per suggestion

tmeasday committed:
Merge pull request #18708 from storybookjs/tom/sb-453-dependencies-ensure-react-dom-mdx

Addon-docs: Move DocsRenderer back to addon-docs

tmeasday committed:
Merge pull request #18730 from storybookjs/add-channel-to-docs-context

Addon-docs: Localize channel to docs context

tmeasday committed:
Make the docs spec more consistently fail (react)

tmeasday committed:
Skip test on react

tmeasday committed:
Skip addon-docs spec for react projects

tmeasday committed:
Skip on CRA too

tmeasday committed:
Merge pull request #18736 from storybookjs/remove-react-18-docs

Don't run the docs spec in `react@18`

ndelangen committed:
Merge branch 'future/base' into future/drop-emotion-10-types

# Conflicts:
#	lib/blocks/src/blocks/DocsContainer.tsx
#	lib/blocks/src/blocks/SourceContainer.tsx
#	lib/blocks/src/blocks/Subtitle.tsx
#	lib/blocks/src/blocks/Title.tsx
#	lib/blocks/src/components/ColorPalette.tsx
#	lib/blocks/src/components/IconGallery.tsx
#	lib/blocks/src/components/Source.tsx
#	lib/blocks/src/components/Story.tsx
#	lib/components/src/placeholder/placeholder.tsx
#	lib/ui/src/components/preview/toolbar.tsx
#	lib/ui/src/components/preview/utils/types.tsx
#	lib/ui/src/components/sidebar/Sidebar.tsx
#	lib/ui/src/components/sidebar/TreeNode.tsx

ndelangen committed:
cleanup

ndelangen committed:
fix

ndelangen committed:
fix

ndelangen committed:
Merge pull request #18598 from storybookjs/drop-emotion-10-types

Future - Drop Emotion 10 types in lib/theming

yannbf committed:
fix(addon-docs): pass remark plugins to mdx loader

shilman committed:
Merge pull request #18740 from storybookjs/yann/sb-414-sb18395-cannot-anchor-to-headings-in-mdx

Addon docs: Pass remarks plugins to mdx loader

tmeasday committed:
Merge remote-tracking branch 'origin/next' into future/base

tmeasday committed:
Shouldn't have dropped this

ndelangen committed:
update snapshots

ndelangen committed:
fix deepscan

Saunved committed:
Build addons/a11y with ts-up

+ Ref issue: https://github.com/storybookjs/storybook/issues/18732

Saunved committed:
Fix entry file names and some import paths

+ There is an issue with the preview file during the linting phase
+ I manually skipped the linter error for now
+ The local demo is running fine

ndelangen committed:
fix

ndelangen committed:
Merge branch 'future/base' into feat/tsup-addons-a11y

ndelangen committed:
remove re-export

ndelangen committed:
Merge pull request #18772 from Saunved/feat/tsup-addons-a11y

Build addons/a11y with ts-up

yannbf committed:
Merge branch 'future/base' into feat/url-in-repro

yannbf committed:
Revert "Merge branch 'future/base' into feat/url-in-repro"

This reverts commit 5edebc1272912dd8ed4082d83c08f85f918159de, reversing
changes made to 1002216c02b4055ddb92d06797a7add6db0e1cf9.

yannbf commented:
This PR borked because of a bad merge using future/base rather than next.

Closing it in favor of https://github.com/storybookjs/storybook/pull/18795
